tests/test_utils.py

```diff
@@ -162,65 +162,85 @@ def check_if_job_finished(job: JobDataclass):
 
 def check_if_job_runner_result_correct(job: JobDataclass):
     """Iterate over every result and check if the distribution of the measurement is correct"""
-    for i in range(len(job.results)):
-        result: ResultDataclass = job.results[i]
-        check_standard_result_data(i, job, result)
-        assert result.meta_data is not None
+
+    assert len(job.results) == 2 * len(job.deployment.programs)
+
+    for result_index in range(len(job.results)):
+        program_index = result_index // 2  # every program is expected to have 2 results
+        result: ResultDataclass = job.results[result_index]
+        check_standard_result_data(program_index, job, result)
+        assert result.meta is not None
         shots: int = job.shots
-        counts: dict = result.result_dict["counts"]
-        probabilities: dict = result.result_dict["probabilities"]
+        data: dict = result.data
 
         # Check if the first result is distributed correctly: 50% for the qubit zero and 50% for the qubit three
-        if (i % 2) == 0:
-            assert compare_values_with_tolerance(shots / 2, counts[BIT_0], COUNTS_TOLERANCE)
-            assert compare_values_with_tolerance(shots / 2, counts[BIT_3], COUNTS_TOLERANCE)
-            assert (counts[BIT_0] + counts[BIT_3]) == shots
-
-            assert compare_values_with_tolerance(PROBABILITY_1 / 2, probabilities[BIT_0], PROBABILITY_TOLERANCE)
-            assert compare_values_with_tolerance(PROBABILITY_1 / 2, probabilities[BIT_3], PROBABILITY_TOLERANCE)
-            assert (probabilities[BIT_0] + probabilities[BIT_3]) > PROBABILITY_1 - PROBABILITY_TOLERANCE
+        if result_index % 4 == 0:
```

Comment:
This check (and the following) is test case specific and should be part of the specific test cases (and not some generic util function...)